### PR TITLE
[ACA-3394] Generally available file preview feature for extensions

### PR DESCRIPTION
**Pull request by popovicsandras**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
See https://issues.alfresco.com/jira/browse/ACA-3394

**What is the new behaviour?**
Done.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -174,17 +174,34 @@ export class AppComponent implements OnInit, OnDestroy {
     const routesWithoutParent = [];
     extensionRoutes.forEach((extensionRoute: ExtensionRoute) => {
       if (this.extensionRouteHasChild(extensionRoute)) {
-        const routeIndex = this.router.config.findIndex(
-          route => route.path === extensionRoute.parentRoute
-        );
-        this.convertExtensionRouteToRoute(extensionRoute);
-        this.router.config[routeIndex].children.unshift(extensionRoute);
+        // const routeIndex = this.router.config.findIndex(
```
Comment (DenysVuika) [inline src/app/app.component.ts:177]: should probably be removed?

File: src/app/app.component.ts [hunk 2]
```diff
@@ -174,17 +174,34 @@ export class AppComponent implements OnInit, OnDestroy {
...
+          parentRoute.children.unshift(extensionRoute);
+        }
       } else {
         routesWithoutParent.push(extensionRoute);
       }
     });
 
     this.router.config.unshift(...routesWithoutParent);
+    this.router.resetConfig(this.router.config);
+    console.log(this.router.config);
```
Comment (DenysVuika) [inline src/app/app.component.ts:196]: note this one should not get merged :)

File: src/app/app.routes.ts [hunk 3]
```diff
@@ -74,6 +74,14 @@ export const APP_ROUTES: Routes = [
       }
     ]
   },
+  {
+    path: 'preview/:nodeId',
+    loadChildren: './components/preview/preview.module#PreviewModule',
+    outlet: 'viewer',
+    data: {
+      navigateSource: '/dev/null'
```
Comment (DenysVuika) [inline src/app/app.routes.ts:82]: hmmm

File: src/app/app.component.ts (hunk 1 above)
Comment (popovicsandras) [reply src/app/app.component.ts:177]: I would like to move the whole logic into the app.extension.service.ts, there is no component related logic here. This PR is half-baked yet (draft), please don't be bothered with it, I just pushed my changes before switching to regression and blocker issues. :)

File: src/app/app.component.ts (hunk 2 above)
Comment (popovicsandras) [reply src/app/app.component.ts:196]: Yepp, it won't, as soon as I finish this PR, I am going to remove the draft mark and ask for a review, please ignore this PR until then :)

File: package.json [hunk 4]
```diff
@@ -16,6 +16,7 @@
     "build": "npm run validate-config && npm run build.app -- --prod",
     "build.e2e": "npm run build.app -- --prod --configuration=e2e",
     "test": "ng test app --code-coverage",
+    "unit": "ng test --browsers=Chrome --watch",
```
Comment (DenysVuika) [inline package.json:19]: I think this is going to be a non-standard command? "npm run unit" ?

Comment (popovicsandras) [reply package.json:19]: @DenysVuika Yes, indeed. It is nothing even needed, I just found it cumbersome to debug the unit tests (in normal chrome with head) in this repo, that is why I created this shortcut command, it has only development aiding purposes.

**Pull request merged**